app/service/data_svc.py

```diff
@@ -179,8 +179,6 @@ async def _load(self, plugins=()):
                 await self._load_abilities(plug)
                 await self._load_objectives(plug)
                 await self._load_packers(plug)
-            await self._verify_ability_set()
-            for plug in plugins:
```

Comment:
183 is there to prevent plugin load order issues. I'm not seeing a change that would ensure that removing 183 wouldn't prevent those issues from happening again

Reply:
Oh I see that now

Reply:
I've changed my mind, I don't believe that I do see the potential plugin load order issues.

I've switched the ordering. Adversaries have objectives and abilities, abilities have payloads. Planners, Sources, and Packers are all independent of these other loads -- they're related to either operations or file downloads.

- Loading payloads shouldn't have any depenencies. '_load_payloads' loads the special payloads and extensions from the `data/payloads/*.yml` files, which don't rely on anything else.
- Abilities don't depend on loaded payloads. During `_load_abilities`, payloads are only stored as a string. The real business happens later in `_verify_abilities`.
- The `_load_objectives` file just loads YAML, those don't depend on anything. These totally self-contained.
- The `_load_adversaries` file just loads YAML, those don't depend on anything. Adversaries store objectives and abilities as strings, objects aren't loaded until they are used.
- The `_load_planners` file just loads YAML, those don't depend on anything.
- The `_load_sources` file just loads YAML, those don't depend on anything. These totally self-contained.
- Packers may have depenencies that are checked in `check_dependencies()`, but that's meant to check for installed programs.

`_load_extensions` and `_verify_data_sets` both have depenencies and call functions that may require all plugins be loaded. These are called after the loop.

I'm hoping you may have more insight into these issues, since they may have been present in the past.

Reply:
I tried verifying this by using tasks in place of every await, hoping that something would clobber another object. There were no issues, but I did uncover a bug where, on a fresh load, abilities are not loaded when the verification functions are run. Fixed in https://github.com/mitre/caldera/pull/1996/commits/b4b5d52d54b34dd42a2416400b45930a2e267436

Reply:
Any update on this? https://github.com/mitre/caldera/commit/b4b5d52d54b34dd42a2416400b45930a2e267436 contains a bug fix which should be added before a release.